x-pack/metricbeat/module/azure/client.go

```diff
@@ -97,24 +171,51 @@ func (client *Client) InitResources(fn mapResourceMetrics) error {
 		client.Log.Debug("no resources were found based on all the configurations options entered")
 	}
 	client.ResourceConfigurations.Metrics = metrics
+
 	return nil
 }
 
-// GetMetricValues returns the specified metric data points for the specified resource ID/namespace.
-func (client *Client) GetMetricValues(metrics []Metric, report mb.ReporterV2) []Metric {
-	var resultedMetrics []Metric
-	// loop over the set of metrics
+// GetMetricValues returns the metric values for the given cloud resources.
+func (client *Client) GetMetricValues(metrics []Metric, reporter mb.ReporterV2) []Metric {
+	var result []Metric
+
+	// Same end time for all metrics in the same batch.
+	referenceTime := time.Now().UTC()
+
 	for _, metric := range metrics {
-		// select period to collect metrics, will double the interval value in order to retrieve any missing values
-		//if timegrain is larger than intervalx2 then interval will be assigned the timegrain value
 		interval := client.Config.Period
-		if t := convertTimegrainToDuration(metric.TimeGrain); t > interval*2 {
-			interval = t
-		}
-		endTime := time.Now().UTC()
-		startTime := endTime.Add(interval * (-2))
+
+		// copy the reference time
+		endTime := referenceTime
+
+		// Fetch in the range [{-2 x INTERVAL},{-1 x INTERVAL}) with a delay of {INTERVAL}.
+		endTime = endTime.Add(interval * (-1))
+		startTime := endTime.Add(interval * (-1))
```

Comment:
Ln 186 to 194 can all move out of the for loop I think?

Reply:
Yes, you're right!

I was convinced that method `Add()` changed the variable's value, but it's false.

I am moving this code block outside the for loop.